data_model/src/block.rs

```diff
@@ -315,24 +315,27 @@ mod committed {
     #[cfg(feature = "std")]
     impl From<&CommittedBlock> for Vec<Event> {
         fn from(block: &CommittedBlock) -> Self {
-            let tx = block
-                .transactions
-                .iter()
-                .cloned()
-                .map(|tx| match &tx.error {
-                    None => PipelineEvent {
-                        entity_kind: PipelineEntityKind::Transaction,
-                        status: PipelineStatus::Committed,
-                        hash: tx.payload().hash().into(),
-                    }
-                    .into(),
-                    Some(error) => PipelineEvent {
-                        entity_kind: PipelineEntityKind::Transaction,
-                        status: PipelineStatus::Rejected(error.clone().into()),
-                        hash: tx.payload().hash().into(),
-                    }
-                    .into(),
-                });
+            let tx = block.transactions.iter().cloned().map(|tx| {
+                tx.error.as_ref().map_or_else(
+                    || {
+                        PipelineEvent {
+                            entity_kind: PipelineEntityKind::Transaction,
+                            status: PipelineStatus::Committed,
+                            hash: tx.payload().hash().into(),
+                        }
+                        .into()
+                    },
+                    |error| {
+                        PipelineEvent {
+                            entity_kind: PipelineEntityKind::Transaction,
+                            status: PipelineStatus::Rejected(error.clone().into()),
+                            hash: tx.payload().hash().into(),
+                        }
+                        .into()
+                    },
+                )
+            });
+
```

Comment:
The exact same piece of code is also changed here:

https://github.com/hyperledger/iroha/pull/3661/files#r1251267503

Not sure which option is better. Just pointing out to the work duplication.

Reply:
Made #3673, should be a good `chore` or `good-first-issue`.